tests/test_slurm_system.py

```diff
@@ -142,3 +143,88 @@ def test_update_node_states_with_mocked_outputs(mock_get_sinfo, mock_get_squeue,
 def test_parse_node_list(node_list: str, expected_parsed_node_list: List[str]):
     parsed_node_list = parse_node_list(node_list)
     assert parsed_node_list == expected_parsed_node_list
+
+
+def setup_mock_slurm_system(slurm_system, group_name):
```

Comment:
`allocate_nodes` doesn't use any field from `SlurmSystem`, it can even be static. Anyway, everything you need for your test function is just `grouped_nodes` and it can be a simple fixture. Let's simplify the setup for these tests, we do not want making it comlicated.